L_ZWay2.lua

```diff
@@ -644,12 +644,18 @@ local command_class = {
   ["50"] = function (d, inst, meta)
     local var = (inst.metrics.scaleTitle or '?'): upper ()
     local translate = {W = "Watts", A = "Amps", V = "Volts"}      -- 2020.02.10 thanks @rafale77
-    if var then setVar (translate[var] or var, inst.metrics.level, meta.service, d) end
+    if var then
+    setVar (translate[var] or var, inst.metrics.level, meta.service, d)
+      if var == "KWH" then
+        setVar ("KWHReading", inst.updateTime, meta.service, d)
+      end
+    end
   end,
 
   -- door lock
   ["98"] = function (d, inst)
       setVar ("Status",open_or_close (inst.metrics.level), SID.DoorLock, d)
+      setVar ("LastTrip", inst.updateTime, SID.SecuritySensor, d)
```

Comment:
This is not quite so simple. Dates/times are an issue.  I worry about the case where openLuup is not running on the same machine as the ZWay hardware and the clocks are not synchronised.  Elsewhere in openLuup I have got around this by always ensuring that the local machine time is used.  It can also be confusing for the Data Historian, because you will get events out of order.

Reply:
Great catch! Didn't think about this.